packages/apollo-gateway/src/FieldSet.ts

```diff
@@ -147,6 +129,38 @@ function mergeSelectionSets(fieldNodes: FieldNode[]): SelectionSetNode {
 
   return {
     kind: 'SelectionSet',
-    selections,
+    selections: mergeFieldNodeSelectionSets(selections),
   };
 }
+
+function mergeFieldNodeSelectionSets(
+  selectionNodes: SelectionNode[],
+): SelectionNode[] {
+  const [fieldNodes, fragmentNodes] = partition(
+    selectionNodes,
+    (node: SelectionNode): node is FieldNode => node.kind === 'Field',
```

Comment:
Sorry I missed this on my first pass. When we check against `node.kind`, we prefer to use (in this case) `Kind.FIELD` rather than string itself by importing `Kind` from graphql-js.